modules/shadertools/src/lib/shader-assembly/resolve-modules.ts

```diff
@@ -38,6 +38,11 @@ function getShaderDependencies(modules: ShaderModuleInstance[]): ShaderModuleIns
     .map(name => moduleMap[name]);
 }
 
+type AbstractModule = {
```

Comment:
Why can't we use type `ShaderModule`? If we need a more minimal type, can we make it a proper export?

Reply:
It seems a heavy type to use for a method that doesn't care about the internals of `ShaderModule`. I will add the export